pkg/migrate/cstor/backup_restore.go

```diff
@@ -0,0 +1,184 @@
+	newBackup := &cstor.CStorBackup{}
+	newBackup.ObjectMeta = metav1.ObjectMeta{
+		Name:        olBackup.Name,
+		Namespace:   olBackup.Namespace,
+		Labels:      olBackup.Labels,
+		Annotations: olBackup.Annotations,
+	}
+	newBackup.Spec = cstor.CStorBackupSpec{
+		BackupName:   olBackup.Spec.BackupName,
+		VolumeName:   olBackup.Spec.VolumeName,
+		SnapName:     olBackup.Spec.SnapName,
+		PrevSnapName: olBackup.Spec.PrevSnapName,
+		BackupDest:   olBackup.Spec.BackupDest,
+		LocalSnap:    olBackup.Spec.LocalSnap,
+	}
+	newBackup.Status = cstor.CStorBackupStatus(olBackup.Status)
+	return newBackup
+}
+
+// TranslateRestoreToV1 translates v1alpha resources to v1
+func TranslateRestoreToV1(oldRestore openebsio.CStorRestore) *cstor.CStorRestore {
+	newRestore := &cstor.CStorRestore{}
+	newRestore.ObjectMeta = metav1.ObjectMeta{
+		Name:        oldRestore.Name,
+		Namespace:   oldRestore.Namespace,
+		Labels:      oldRestore.Labels,
+		Annotations: oldRestore.Annotations,
+	}
+	newRestore.Spec = cstor.CStorRestoreSpec{
+		RestoreName:   oldRestore.Spec.RestoreName,
+		VolumeName:    oldRestore.Spec.VolumeName,
+		RestoreSrc:    oldRestore.Spec.RestoreSrc,
+		MaxRetryCount: oldRestore.Spec.MaxRetryCount,
+		RetryCount:    oldRestore.Spec.RetryCount,
+		StorageClass:  oldRestore.Spec.StorageClass,
+		Size:          oldRestore.Spec.Size,
+		Local:         oldRestore.Spec.Local,
+	}
+	newRestore.Status = cstor.CStorRestoreStatus(oldRestore.Status)
+	return newRestore
+}
+
+// TranslateCompletedBackupToV1 translates v1alpha resources to v1
+func TranslateCompletedBackupToV1(oldCompletedBackup openebsio.CStorCompletedBackup) *cstor.CStorCompletedBackup {
+	newCompletedBackup := &cstor.CStorCompletedBackup{}
+	newCompletedBackup.ObjectMeta = metav1.ObjectMeta{
+		Name:        oldCompletedBackup.Name,
+		Namespace:   oldCompletedBackup.Namespace,
+		Labels:      oldCompletedBackup.Labels,
+		Annotations: oldCompletedBackup.Annotations,
+	}
+	newCompletedBackup.Spec = cstor.CStorCompletedBackupSpec{
+		BackupName:         oldCompletedBackup.Spec.BackupName,
+		VolumeName:         oldCompletedBackup.Spec.VolumeName,
+		LastSnapName:       oldCompletedBackup.Spec.PrevSnapName,
+		SecondLastSnapName: oldCompletedBackup.Spec.SnapName,
+	}
+	return newCompletedBackup
+}
+
+func (c *CSPCMigrator) upgradeBackupRestore(cspUID string, cspiObj *cstor.CStorPoolInstance) error {
+	// Migrate backup to v1 version
+	oldBackupList, err := c.OpenebsClientset.OpenebsV1alpha1().
+		CStorBackups(c.OpenebsNamespace).List(metav1.ListOptions{
+		LabelSelector: cspUIDLabel + "=" + cspUID,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "failed to list v1alpha1 cstorbackups")
+	}
+	for _, oldBackup := range oldBackupList.Items {
+		newBackup := TranslateBackupToV1(oldBackup)
+		newBackup.Labels[types.CStorPoolInstanceNameLabelKey] = cspiObj.Name
+		newBackup.Labels[types.CStorPoolInstanceUIDLabelKey] = string(cspiObj.UID)
```

Comment:
The CSP labels will be preserved, right? 
Is this intentional? Do we plan to put the previous CSP label by any other key to the current way is fine?

Reply:
Ok. Never mind. We are deleting the old labels...  Ignore it.